### PR TITLE
Add scheduler function that schedules for a range of days

### PR DESCRIPTION
**Pull request by fuiszgt**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] PR has a descriptive title
- [x] PR contains closely related changes only (should not have many unrelated changes)
- [x] STEP pod members have been added as reviewers
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

### CODE REVIEW

**Comment by fuiszgt** [comment]
Sorry, this pull request is made by accident. I still need to add tests.

File: src/main/java/com/google/sps/scheduler/Scheduler.java [hunk 1]
```diff
@@ -46,7 +46,17 @@
    * For each task that can be scheduled, the due time is set and a list of all scheduled tasks
    * is returned.
    */
-  public static List<Task> schedule(
+  public static List<Task> scheduleInRange(List<Event> calendarEvents, List <Task> tasks, String timeZone, LocalDate startDate, LocalDate endDate){
```
Comment (dola) [inline src/main/java/com/google/sps/scheduler/Scheduler.java:49]: nit: `List<Task>` not `List <Task>`

File: src/main/java/com/google/sps/scheduler/Scheduler.java [hunk 2]
```diff
@@ -46,7 +46,17 @@
...
+    LocalDate scheduleDate = startDate;
+    List<Task> scheduledTasks = new ArrayList<>();
+    while(0 >= scheduleDate.compareTo(endDate)){
```
Comment (dola) [inline src/main/java/com/google/sps/scheduler/Scheduler.java:52]: There is isAfter and isBefore as well: https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html#isAfter-java.time.chrono.ChronoLocalDate-

**Comment by dola** [comment]
btw you can close a PR again so it doesn't show up until you are ready for it to be review

**Review by Giost — APPROVED** [review]
LGTM

File: src/main/java/com/google/sps/scheduler/Scheduler.java [hunk 3]
```diff
@@ -46,7 +46,17 @@
    * For each task that can be scheduled, the due time is set and a list of all scheduled tasks
    * is returned.
    */
-  public static List<Task> schedule(
+  public static List<Task> scheduleInRange(List<Event> calendarEvents, List<Task> tasks, String timeZone, LocalDate startDate, LocalDate endDate) {
+    LocalDate scheduleDate = startDate;
+    List<Task> scheduledTasks = new ArrayList<>();
+    while (!scheduleDate.isAfter(endDate)) {
```
Comment (zlorf) [inline src/main/java/com/google/sps/scheduler/Scheduler.java:52]: The condition could also check if there are any tasks to be scheduled left.

File: src/test/java/com/google/sps/scheduler/SchedulerTest.java [hunk 4]
```diff
@@ -368,4 +369,181 @@ public void overlappedEventsDifferentTimeZonesAndDays() {
...
+    calendarEvents.add(eventF);
+    calendarEvents.add(eventE);
+    calendarEvents.add(eventD);
+    calendarEvents.add(eventC);
+    calendarEvents.add(eventB);
+    calendarEvents.add(eventA);
+    calendarEvents.add(eventG);
+    calendarEvents.add(eventJ);
+
+    List<Task> actualScheduledTasks = scheduleInRange(calendarEvents, sampleTasks, ZURICH_TIME_ZONE, day, nextDay);
```
Comment (zlorf) [inline src/test/java/com/google/sps/scheduler/SchedulerTest.java:430]: Please make the endpoint far in the future, like 2 years ahead.

**Comment by zlorf** [comment]
Nice test case!

**Review by zlorf — APPROVED** [review]
Left a few minor comments.

**Comment by fuiszgt** [comment]
Alternative considered when rewriting the scheduler:
Instead of creating a new scheduleInRange() function, we could have treated the multiple day interval as a single interval, and added "Mock" events to block out time in the early morning and late evening to prevent scheduling to that time. This more general solution could have been used in a larger project where more future improvements are expected, but we deemed our solution of iterating the day-scheduling function simpler for our case.

**Pull request merged**